cmd/appengine/device.go

```diff
@@ -959,6 +960,33 @@ func devicesSendDataF(command *cobra.Command, args []string) error {
 	return nil
 }
 
+func shouldSkipRealmManagementChecks(cmd cobra.Command) (bool, error) {
+	skipRealmManagementChecks, err := cmd.Flags().GetBool("skip-realm-management-checks")
+	if err != nil {
+		return false, err
+	}
+
+	if skipRealmManagementChecks || astarteAPIClient.RealmManagement == nil {
+		return true, nil
+	} else {
+		token, err := cmd.Flags().GetString("token")
+		if err != nil {
+			return false, err
+		}
+
+		hasRealmManagementClaim, err := misc.IsJWTAstarteClaimValidForService(token, misc.RealmManagement)
```

Comment:
Maybe add a short comment above each block of checks to explain what is going on? For example here at a first sight it's not clear why if this returns an error you don't bubble it up but you return `false, nil`